src/lib.rs

```diff
@@ -0,0 +1,222 @@
+use pyo3::ffi::{
+    PyContextVar_Get, PyEval_SetProfile, PyFrameObject, PyUnicode_AsUTF8, Py_tracefunc,
+    CO_ASYNC_GENERATOR, CO_COROUTINE, CO_ITERABLE_COROUTINE,
+};
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use pyo3::{AsPyPointer, PyAny, Python};
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use std::ffi::CStr;
+use std::os::raw::c_int;
+use std::time::Instant;
+
+/// Enum of possible Python's Trace/Profiling events
+#[allow(dead_code)]
+#[repr(i32)]
+enum TraceEvent {
+    Call,
+    Exception,
+    Line,
+    Return,
+    CCall,
+    CException,
+    CReturn,
+    Opcode,
+    __NonExhaustive,
```

Comment:
`#[non_exhaustive]` is stable, you can use it on `TraceEvent` instead of the `__NonExhaustive` variant trick